bugsnag-android-unity/src/main/java/com/bugsnag/android/unity/UnityClient.java

```diff
@@ -10,32 +10,115 @@
 
 public class UnityClient {
 
+    static Client client;
+
     public static void init(Context androidContext, String apiKey, boolean trackSessions) {
+        if (client != null) {
+            client.disableExceptionHandler();
```

Comment:
My inclination would be to no op the method if client is non null. We could use an atomicboolean to ensure the method is only run once to also solve @fractalwrench's comment below? I think this will also mean that the config changes persist properly. Only downside as @martin308 says is that any changes to apikey or tracksessions by calling init multiple times would not be possible - but personally that seems both unlikely and a reasonable api restriction.

Thoughts?

Reply:
@martin308 I think a couple of the above points are also applicable to the `Register` method in `Bugsnag.cs`. I'm not too familiar with how C# handles concurrency - would you be able to have a look and see whether there are any potential issues?